### PR TITLE
[documentation] Highlight difference across flow in PF/pipeline job

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Update flow in pipeline doc on the difference.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: docs/cloud/azureai/use-flow-in-azure-ml-pipeline.md [hunk 1]
```diff
@@ -127,3 +127,9 @@ ml_client.jobs.create_or_update(pipeline_with_flow)
 :::
 
 ::::
+
+## Difference across flow in prompt flow and pipeline job
+
+In prompt flow, flow runs on runtime (a computing resource usually hosted on a compute instance), which can be regarded exclusive for flow run; while in pipeline job, flow runs on the configured general compute resources (usually compute cluster).
```
Comment (wangchao1230) [inline docs/cloud/azureai/use-flow-in-azure-ml-pipeline.md:133]: what do you mean by:   regarded exclusive for flow run?

Comment (wangchao1230) [inline docs/cloud/azureai/use-flow-in-azure-ml-pipeline.md:133]: may add a link to runtime or compute doc here

Comment (zhengfeiwang) [reply docs/cloud/azureai/use-flow-in-azure-ml-pipeline.md]: offline synced on this, and updated the wording

Comment (zhengfeiwang) [reply docs/cloud/azureai/use-flow-in-azure-ml-pipeline.md]: added one link

**Review by wangchao1230 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**